spec/support/api_url_helper.rb

```diff
@@ -70,26 +71,60 @@ def stub_api_modify(*components, body:, payload:, action: :post, status: 201)
     def stub_api_put(*components, body:, payload:)
       stub_api_modify(*components, action: :put, status: 200, body: body, payload: payload)
     end
+  end
 
-    def stub_api_v2_post(klass)
-      # intercepts the 'update' method for any class beginning with
-      # 'Sequencescape::Api::V2::' and returns true
+  module V2Helpers
+    def stub_api_v2_patch(klass)
+      # intercepts the 'update' and 'update!' method for any instance of the class beginning with
+      # 'Sequencescape::Api::V2::' and returns true.
       receiving_class = "Sequencescape::Api::V2::#{klass}".constantize
       allow_any_instance_of(receiving_class).to receive(:update).and_return(true)
+      allow_any_instance_of(receiving_class).to receive(:update!).and_return(true)
     end
 
     def stub_api_v2_save(klass)
-      # intercepts the 'save' method for any class beginning with
-      # 'Sequencescape::Api::V2::' and returns true
+      # intercepts the 'save' method for any instance of the class beginning with
+      # 'Sequencescape::Api::V2::' and returns true.
       receiving_class = "Sequencescape::Api::V2::#{klass}".constantize
       allow_any_instance_of(receiving_class).to receive(:save).and_return(true)
     end
 
+    def stub_api_v2_post(klass, return_value = nil)
+      # intercepts the 'create!' method for any class beginning with
+      # 'Sequencescape::Api::V2::' and returns the given value or else true.
+      receiving_class = "Sequencescape::Api::V2::#{klass}".constantize
+      return_value ||= true
+      allow(receiving_class).to receive(:create!).and_return(return_value)
```

Comment:
Does it need to intercept any `create`  (or `save` above) without the bang too?

Reply:
I did intercept `create` without the`!` and it never gets called in our code so I stopped doing it.  I suspect I'll find some case soon that needs it, but I didn't see the point in stubbing something we're not calling.  I will be easy to add if it's needed at the time.  I have this file open as a permanent fixture during the Limber migration.